lib/AST/DistributedDecl.cpp

```diff
@@ -271,12 +271,24 @@ swift::getDistributedSerializationRequirements(
   if (existentialRequirementTy->isAny())
     return true; // we're done here, any means there are no requirements
 
-  auto serialReqType = existentialRequirementTy->castTo<ExistentialType>()
-                           ->getConstraintType()
-                           ->getDesugaredType();
+  if (auto alias = dyn_cast<TypeAliasType>(existentialRequirementTy.getPointer())) {
+    auto ty = alias->getDesugaredType();
+    if (isa<ClassType>(ty) || isa<StructType>(ty) || isa<EnumType>(ty)) {
+      // SerializationRequirement cannot be class or struct nowadays
```

Comment:
I think we should explain this better - it cannot be anything other then a protocol because it would be used to constrain a generic parameter associated with serialization methods.

Reply:
And might be check for `!ty->is<ProtocolType>()` to make it simpler :)

Reply:
I think you should use `!ty->isConstraintType()` which also includes protocol composition types and was added for exactly this purpose 🙂 I think the code right now also doesn't handle bound generic types, e.g. `BoundGenericClassType`, so checking specifically for constraint types would fix that.

Reply:
Yeah, `isConstraintType` is the best option here.

Reply:
I forgot that in the conformance it's written as an existential type, so this should actually check for `isExistentialType()` I think

Reply:
Great, thanks for the info!

Reply:
I'm still confused by this code. Why are we explicitly looking for a `TypeAliasType` here rather than doing a purely semantic check for "is existential type" and then verifying that the only requirements are protocol requirements?

Reply:
Sorry folks for the delay, back to it now after the WWDC crunch 🙏 
Will send in a PR to clean this up according to comments -- thanks @hborla !